### PR TITLE
E2E Tests: Add test around cross-origin isolation 

### LINKED ISSUE #7099: E2E Tests: Add test around cross-origin isolation
## Task Description

Would be good to have a e2e test to verify:

* Cross-origin isolation is enabled (`window.crossOriginIsolated === true`) when video optimization is enabled
* Things like 3P media still work (see #7016)  
  Could do something simple like:  
  1. Turn on video optimization
  2. Go to 3P media tab
  3. Insert image from each provider  
  If there are problems there would be console errors which we get notified  about. No snapshot because those images always change.

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Add a simple e2e test to insert, from unsplash, tenor and coverr. 

## Relevant Technical Choices
This PR add some simple workaround for accessibility issues. These include. 

- Changing the tenor logo to name of the brand and removing powered by. https://github.com/google/web-stories-wp/issues/7274
- At a simple alt to 3P assets.  https://github.com/google/web-stories-wp/issues/7273

The first test includes a test to dismiss 3p agreement. This is required. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7099

### CODE REVIEW

File: assets/src/edit-story/app/media/media3p/attribution.js [hunk 1]
```diff
@@ -110,7 +110,8 @@ export function TenorAttribution() {
   return (
     <a href={tenorUrl} target={'_blank'} rel={'noreferrer'}>
       <AttributionPill>
-        <TenorLogo data-label={__('Powered by Tenor', 'web-stories')} />
+        {__('Powered by', 'web-stories')}
+        <TenorLogo />
```
Comment (swissspidy) [inline assets/src/edit-story/app/media/media3p/attribution.js]: Why was this changed?

Comment (swissspidy) [reply assets/src/edit-story/app/media/media3p/attribution.js]: #7358 is fixing this already

**Review by swissspidy — CHANGES_REQUESTED** [review]
As per the ticket description, there should be a at least a new test that checks whether `window.crossOriginIsolated` is true.

File: assets/src/edit-story/app/media/media3p/attribution.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/media/media3p/attribution.js]: Yeah this is a work around for #7358. That PR wasn't an thing when I started ticket 12 days ago.

**Comment by spacedmonkey** [comment]
> As per the ticket description, there should be a at least a new test that checks whether window.crossOriginIsolated is true.

Why wouldn't it be true? Why is there a test in e2e for this. Really confused by this.

File: assets/src/edit-story/app/media/media3p/attribution.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/app/media/media3p/attribution.js]: Let's remove this in favor of said PR then.

**Comment by swissspidy** [comment]
> > As per the ticket description, there should be a at least a new test that checks whether window.crossOriginIsolated is true.
> 
> Why wouldn't it be true? Why is there a test in e2e for this. Really confused by this.

This way we could spot any bugs where the headers aren't correctly set. It should be like 1 line of JS to add which doesn't cause any harm and verifies that cross-origin isolation is enabled.

File: assets/src/edit-story/app/media/media3p/attribution.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/media/media3p/attribution.js]: Okay, the e2e tests fail until #7358 is merged. I have marked it as a blocker..

**Comment by spacedmonkey** [comment]
> This way we could spot any bugs where the headers aren't correctly set. It should be like 1 line of JS to add which doesn't cause any harm and verifies that cross-origin isolation is enabled.

Something like this then?
```
const crossOriginIsolated = await page.evaluate(() => window.crossOriginIsolated;
expect(crossOriginIsolated).toBeTrue();
```

**Comment by swissspidy** [comment]
> > This way we could spot any bugs where the headers aren't correctly set. It should be like 1 line of JS to add which doesn't cause any harm and verifies that cross-origin isolation is enabled.
> 
> Something like this then?
> 
> ```
> const crossOriginIsolated = await page.evaluate(() => window.crossOriginIsolated;
> expect(crossOriginIsolated).toBeTrue();
> ```

Yup 👍

**Comment by spacedmonkey** [comment]
```
const crossOriginIsolated = await page.evaluate(() => window.crossOriginIsolated);
expect(crossOriginIsolated).toBeTrue();
```

I couldn't get this working. I think it is related to #7328 . Loading an older version of the browser and puppeteer

File: assets/src/edit-story/app/media/media3p/attribution.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/app/media/media3p/attribution.js]: Why are they failing? Because of aXe? Then disable the aXe rule until that PR is merged.

File: packages/e2e-tests/src/specs/editor/media/insert3PMedia.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import { createNewStory } from '@web-stories-wp/e2e-test-utils';
+
+describe('Inserting 3P Media', () => {
+  it('should dismiss message', async () => {
+    await createNewStory();
+
+    const crossOriginIsolated = await page.evaluate(
+      () => window.crossOriginIsolated
+    );
+    expect(crossOriginIsolated).toBeTrue();
```
Comment (swissspidy) [inline packages/e2e-tests/src/specs/editor/media/insert3PMedia.js]: IMO this better fits into `packages/e2e-tests/src/specs/editor/editor.js` in a new separate test.

It is not related to 3P media in any way

Comment (spacedmonkey) [reply packages/e2e-tests/src/specs/editor/media/insert3PMedia.js]: Moved in https://github.com/google/web-stories-wp/pull/7272/commits/76449cb10808932bf1bd72969c5e5b7d9d3f7800

File: packages/e2e-tests/src/specs/editor/media/insert3PMedia.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { createNewStory } from '@web-stories-wp/e2e-test-utils';
+
+describe('Inserting 3P Media', () => {
```
Comment (swissspidy) [inline packages/e2e-tests/src/specs/editor/media/insert3PMedia.js:21]: These are very useful tests to have! Nice work 👍

File: packages/e2e-tests/src/specs/editor/editor.js [hunk 4]
```diff
@@ -42,4 +42,13 @@ describe('Story Editor', () => {
...
   });
+
+  it('should cross origin isolation be enabled in the browser', async () => {
+    await createNewStory();
+
+    const crossOriginIsolated = await page.evaluate(
+      () => window.crossOriginIsolated
+    );
+    expect(crossOriginIsolated).toBeTrue();
+  });
```
Comment (swissspidy) [inline packages/e2e-tests/src/specs/editor/editor.js]: Now that I think about it, it would be very beneficial to have a separate test in addition to this one were:

1. User goes to Settings
2. User disables video optimization
3. User goes to editor
4. `window.crossOriginIsolated` is `false`

That'd really give us some assurance that it will be disabled as expected, which is what we communicate in support.

File: packages/e2e-tests/src/specs/editor/editor.js [hunk 5]
```diff
@@ -42,4 +42,13 @@ describe('Story Editor', () => {
     await percySnapshot(page, 'Empty Editor on RTL');
     await deactivateRTL();
   });
+
+  it('should cross origin isolation be enabled in the browser', async () => {
```
Comment (swissspidy) [inline packages/e2e-tests/src/specs/editor/editor.js]:
```suggestion
  it('should have cross-origin isolation enabled', async () => {
```

File: packages/e2e-tests/src/specs/editor/editor.js [hunk 6]
```diff
@@ -22,8 +22,19 @@ import {
...
 } from '@web-stories-wp/e2e-test-utils';
 
+async function toggleVideoOptimization() {
+  await visitSettings();
+  const selector = '[data-testid="media-optimization-settings-checkbox"]';
+  await page.waitForSelector(selector);
+  // Clicking will only act on the first element.
+  await expect(page).toClick(selector);
+  // Await REST API request.
+  await page.waitForTimeout(1000);
```
Comment (swissspidy) [inline packages/e2e-tests/src/specs/editor/editor.js]: It's recommended to use `waitForResponse` to await requests like this, e.g.

```js
await page.waitForResponse((response) =>
  response.url().includes('web-stories/v1/users/me')
);
```

Comment (spacedmonkey) [reply packages/e2e-tests/src/specs/editor/editor.js]: Sadly it doesn't work :(  100% of the time https://github.com/google/web-stories-wp/runs/2546792286#step:15:15 . 

I have added the one second wait back. it is sad, the there is viasual queue to say it has saved we can look for it code.

Comment (swissspidy) [reply packages/e2e-tests/src/specs/editor/editor.js]: Why doesn't this work? Are there perhaps two HTTP requests being made? Then we should add a second `waitForResponse` with more specific predicates.

But we should really, really avoid such timeouts when there are better alternatives available to us.

Comment (swissspidy) [reply packages/e2e-tests/src/specs/editor/editor.js]: Still not working with b0ca796ea49bcca3d7e4142994074f795683d6fb

Is it working for you locally? Or is it a an actual bug?

Comment (spacedmonkey) [reply packages/e2e-tests/src/specs/editor/editor.js]: It seems like it is a bug in 5.3/5.4. Default values for meta data was only added in 5.5. Meaning the default value of true for user meta is not respected. I have added a workaround.

**Comment by swissspidy** [comment]
`Inserting 3P Media › should insert an coverr video` seems to be failing now

**Comment by spacedmonkey** [comment]
> Inserting 3P Media › should insert an coverr video seems to be failing now

I know :(

I can't debug it locally. That okay to disable that test for now?

**Comment by swissspidy** [comment]
> > Inserting 3P Media › should insert an coverr video seems to be failing now
> 
> I know :(
> 
> I can't debug it locally. That okay to disable that test for now?

Sure, but please create a ticket to fix it, and mention that ticket in a code comment.

**Comment by swissspidy** [comment]
Some more failing E2E tests here, but they work fine on `main` and also other PRs it seems.

So.. a bug?

**Comment by spacedmonkey** [comment]
> So.. a bug?

I honestly don't know. I can't work it out. I have run the tests like 5 times and get different results each time. Debug this is almost impossible.

**Comment by spacedmonkey** [comment]
@swissspidy I have run these test 3 times in a row and they work for me now. They also work locally. 


I would really like to see the fit for default values find it's way into 1.7...

File: packages/e2e-tests/src/specs/editor/media/insert3PMedia.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  it('should dismiss message', async () => {
+    await createNewStory();
+
+    await expect(page).not.toMatchElement('[data-testid="FrameElement"]');
+    await expect(page).toClick('#library-tab-media3p');
+    await expect(page).toClick('button', { text: 'Dismiss' });
```
Comment (swissspidy) [inline packages/e2e-tests/src/specs/editor/media/insert3PMedia.js:27]: Nit: this only works if tests are executed in order and if this is the first media3p test to run. So not 100% bulletproof.
